completed the popup window and refactored the related product widgets by using data provided by the server

DevonL1010 opened the pull request.

merosenlund commented on src/components/related/ProductCardList.jsx:
```diff
@@ -3,35 +3,40 @@ import axios from 'axios';
     <div className={styles.productCardList}>
-      {relatedID.map((result, index) => {
+      {Object.values(related).map((result, index) => {
+        // console.log(result)
```
You have a lot of commented out code throughout these files and it makes it a little bit hard to follow. If you have a section of code to the point that it is working it would be nice to get rid of all console.logs. 

As far as the other commented out code goes I am kind of the same way and don't like getting rid of code in case I need it again in the future. I have been trying to force myself to not allow this to happen though because it makes it harder to follow. I would encourage you to look through your commented out code and remove any that you are pretty sure you won't need to use anymore.

Another option for code that you won't need to use again for the current project but you want to remember how you did it you could copy and paste to your engineering journal.

DevonL1010 replied:
The comment-out code is kept for possible future optimization. I think the way we grouping all the data together and then send to the client is not really efficient. I will move all the comment-out code to the very button.

merosenlund replied:
You are right there is definitely lots of room to improve the server code and optimize how we want to get the data we need.

merosenlund reviewed (changes requested):
Hey Devon, overall this looks pretty good. There are some questions I have though and it looks like this has quite a bit of commented out code that could be cleaned up. 

I think in general we want to have the code we are merging with main be as clean as possible because that is our 'production' code. However if some of the commented code you think you are going to need for future features feel free and keep it in.

I'm also working on my review skills so please give me feedback on if I'm being to picky or if I'm misunderstanding the intent behind any of your code.

DevonL1010 commented on server/index.js:
```diff
@@ -69,13 +72,13 @@ app.get('/products', (req, res) => {
       }
-      res.send(200, product);
+      // res.send(200, product);
+      res.status(200).send(product)
```
vscode marks it as  error and suggests rewriting in this way.

merosenlund replied:
Sounds good! We can just delete the commented out way then. Thanks for fixing this!

Do you have eslint set up? I need to get that working still.

DevonL1010 commented on server/index.js:
```diff
@@ -37,6 +38,8 @@ app.get('/products', (req, res) => {
     product.cart = result[6].data;
     product.answers = {};
     product.related = {};
+    // console.log(product)
```
sure.

DevonL1010 commented on src/components/related/OutfitList.jsx:
```diff
@@ -1,8 +1,24 @@
+  }, []);
   return (
     <div className={styles.OutfitList}>
       <i class="fas fa-plus">Add to Outfit</i>
```
sure

DevonL1010 commented on src/components/related/ProductCard.jsx:
```diff
@@ -21,43 +25,40 @@ const ProductCard = ({overviewProduct, relatedID}) =>{
 
     //get related product image
-    axios.get(`${API_URL}/${relatedID}/styles`, options)
+    axios.get(`${API_URL}/${relatedProduct.info.id}/styles`, options)
```
I thought the image data was not available. I will update the code. Thanks!

DevonL1010 commented:
- moved all the comment-out code to the very button 
- change the 'class' to 'className'
- moved all the axios request and using useContext instead

merosenlund reviewed (approved):
Looks great! Thanks for making the changes, sorry if I was being too picky.

DevonL1010 commented:
That is a TRUE code review! Appreciate all your helps.
